hypershift-operator/controllers/scheduler/dedicated_request_serving_nodes.go

```diff
@@ -257,3 +272,366 @@ func (r *DedicatedServingComponentScheduler) Reconcile(ctx context.Context, req
+		// node affinity configuration on the deployment
+		return false, newDeployment(placeholderNamespace, placeholderSize, parsedIndex, pairedNodesAssignedToHostedClusters.UnsortedList()), nil
+	}
+	return false, nil, nil
+}
+
+func newDeployment(namespace, sizeClass string, placeholderIndex int, pairedNodes []string) *appsv1applyconfigurations.DeploymentApplyConfiguration {
+	return appsv1applyconfigurations.Deployment(strconv.Itoa(placeholderIndex), namespace).WithSpec(
+		appsv1applyconfigurations.DeploymentSpec().
+			WithReplicas(2).
+			WithSelector(metav1applyconfigurations.LabelSelector().WithMatchLabels(map[string]string{
+				PlaceholderLabel: strconv.Itoa(placeholderIndex),
+				hostedclustersizing.HostedClusterSizeLabel: sizeClass,
+			})).
+			WithTemplate(corev1applyconfigurations.PodTemplateSpec().
+				WithLabels(map[string]string{
+					PlaceholderLabel: strconv.Itoa(placeholderIndex),
+					hostedclustersizing.HostedClusterSizeLabel: sizeClass,
+				}).
+				WithSpec(corev1applyconfigurations.PodSpec().
+					// placeholder pods must land on request serving nodes for the size class we're keeping warm
+					WithNodeSelector(map[string]string{
+						ControlPlaneServingComponentLabel:          "true",
+						hostedclustersizing.HostedClusterSizeLabel: sizeClass,
+					}).
+					WithAffinity(corev1applyconfigurations.Affinity().
+						WithPodAffinity(corev1applyconfigurations.PodAffinity().WithRequiredDuringSchedulingIgnoredDuringExecution(
+							// placeholder pods must land on nodes with matching paired-nodes labels
+							corev1applyconfigurations.PodAffinityTerm().WithLabelSelector(metav1applyconfigurations.LabelSelector().WithMatchExpressions(
+								metav1applyconfigurations.LabelSelectorRequirement().
+									WithKey(PlaceholderLabel).
+									WithOperator(metav1.LabelSelectorOpIn).
+									WithValues(strconv.Itoa(placeholderIndex)),
+							)).WithTopologyKey(OSDFleetManagerPairedNodesLabel),
+						)).
+						WithPodAntiAffinity(corev1applyconfigurations.PodAntiAffinity().WithRequiredDuringSchedulingIgnoredDuringExecution(
+							// placeholder pods must land in different zones
+							corev1applyconfigurations.PodAffinityTerm().WithLabelSelector(metav1applyconfigurations.LabelSelector().WithMatchExpressions(
+								metav1applyconfigurations.LabelSelectorRequirement().
+									WithKey(PlaceholderLabel).
+									WithOperator(metav1.LabelSelectorOpIn).
+									WithValues(strconv.Itoa(placeholderIndex)),
+							)).WithTopologyKey("topology.kubernetes.io/zone"),
+							// placeholders pods for this deployment can't land on a node where other placeholders already exist
+							corev1applyconfigurations.PodAffinityTerm().WithLabelSelector(metav1applyconfigurations.LabelSelector().WithMatchExpressions(
+								metav1applyconfigurations.LabelSelectorRequirement().
+									WithKey(PlaceholderLabel).
+									WithOperator(metav1.LabelSelectorOpNotIn).
+									WithValues(strconv.Itoa(placeholderIndex)),
```

Comment:
The hostname antiaffinity should be over a label/value that all placeholder pods share, not just the ones at a particular index.

Reply:
Oops, good catch - this should not have any value.